File: src/main/java/failchat/handlers/HtmlHandler.java
package failchat.handlers;

import failchat.core.Message;
import failchat.core.MessageHandler;

/**
 * Заменяет символы '<' и '>' на character entities
 * */
public class HtmlHandler implements MessageHandler {
    @Override
    public void handleMessage(Message message) {
        String messageText = message.getText();
        messageText = messageText.replaceAll("<", "&lt;");
        messageText = messageText.replaceAll(">", "&gt;");
        message.setText(messageText);
    }
}

File: src/main/kotlin/failchat/core/chat/handlers/HtmlHandler.kt
package failchat.core.chat.handlers

import failchat.core.chat.ChatMessage
import failchat.core.chat.MessageHandler

/**
 * Заменяет символы '<' и '>' на html character entities.
 */
class HtmlHandler : MessageHandler<ChatMessage> {
    override fun handleMessage(message: ChatMessage) {
        message.text = message.text.apply {
            replace("<", "&lt;")
            replace(">", "&gt;")
        }
    }
}
